src/tuple_impl.rs

```diff
@@ -63,7 +63,7 @@ impl<T> Iterator for TupleBuffer<T>
         } else {
             buffer.iter()
                   .position(|x| x.is_none())
-                  .unwrap_or_else(|| buffer.len())
+                  .unwrap_or(buffer.len())
```

Comment:
Out of curiosity: Was this suggested by `clippy`? If not, I'd stay with the original

Reply:
Yes: https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_lazy_evaluations

But there's an open issue for that, because I find that strange too: https://github.com/rust-lang/rust-clippy/issues/8109

Reply:
I could not find authoritative reference that calling `len` is guaranteed to only read a field, so please revert this to `unwrap_or_else`, and possibly add a lint.